### PR TITLE
Simpler Dockerfiles, with more platforms

### PR DESCRIPTION
**Pull request by gquintard**
- add ubuntu and centos images
- leverage the upstream packages for rpm and deb platforms
- for alpine, us the aports
- simplify update.sh

### CODE REVIEW

**Comment by teohhanhui** [comment]
> add ubuntu and centos images

Unnecessary. Docker official images only use Debian and Alpine.

**Comment by gquintard** [comment]
> Unnecessary. Docker official images only use Debian and Alpine.

thanks, I didn't know that, I'll nuke them then

**Comment by teohhanhui** [comment]
Sorry, I'm gonna close this. This is not gonna get us any closer to having an official Docker image. It has too many problems.

**Comment by gquintard** [comment]
Can you explain which ones?

**Comment by teohhanhui** [comment]
~But I'll take a look at the aports / `APKBUILD` and see if we could use that.~

I guess not.

File: Dockerfile-debian.template [hunk 1]
```diff
@@ -1,105 +1,16 @@
...
 
-# add our user and group first to make sure their IDs get assigned consistently, regardless of whatever dependencies get added
-RUN set -eux; \
-	groupadd -r varnish; \
-	for user in varnish vcache; do \
-		useradd -r -g varnish $user; \
-	done
+RUN apt-get update && \
+    apt-get install -y curl && \
+    curl -s https://packagecloud.io/install/repositories/varnishcache/varnish%%VARNISH_VERSION%%/script.deb.sh |  bash && \
```
Comment (teohhanhui) [inline Dockerfile-debian.template:5]: Take for example, this part. We need (mostly) reproducible builds.

https://github.com/docker-library/official-images/blob/master/README.md#repeatability

**Comment by gquintard** [comment]
Easy to change:

```
sudo apt-get install debian-archive-keyring curl gnupg apt-transport-https
curl -L https://packagecloud.io/varnishcache/varnish62/gpgkey | apt-key add -
echo deb https://packagecloud.io/varnishcache/varnish62/ubuntu/ trusty main > /etc/apt/sources.list.d/varnish.list
```

we can even embark the gpgkey so we don't have to download it. `script.deb.sh` is a helper, we can avoid using it, I was just trying to showcase how easy it was to just use the repo.

> https://github.com/docker-library/official-images/blob/master/README.md#repeatability

so your issue is not really about the fact that we are using `script.deb.sh`, but that packages aren't pinned? If so, even we the change above, the file installs 5 packages, they are easy to lock down.

Would it make you happier if we got the package source from packagecloud and if we rebuilt it locally? If the alpine approach works for you, we can do the same thing with debian. I believe it's a waste of CPU power, but you get your package built from source, I get simpler Dockerfiles, and we all get proper packages

**Comment by teohhanhui** [comment]
That's what we already do in #42? :smile:

I don't build from source anymore for Debian, except for multiarch.

**Comment by gquintard** [comment]
To be honest, I have no idea what happens it #42. the code is crazy

**Comment by teohhanhui** [comment]
It's adapted from https://github.com/docker-library/postgres/blob/master/Dockerfile-debian.template

It's the one @tianon pointed us to. So we're in good hands. As official as it can get, learning from well-made Docker official images.

**Comment by gquintard** [comment]
What about this: https://gist.github.com/gquintard/e3fd273388d7d685338306f975b9b73d ?

I don't understand why you would go for the monstrosity that is https://github.com/docker-library/postgres/blob/master/Dockerfile-debian.template when you can have a maintainable and legible Dockerfile.

I'm far from being a Docker genius, but the `Dockerfile` I linked to, I understand, `Dockerfile-debian.template` not so much, which is the reason I can't/won't review it.

**Pull request closed without merging**